flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/TaskMailbox.java

```diff
@@ -225,4 +226,13 @@ public boolean isAcceptingMails() {
 	 * @param runnable the runnable to execute
 	 */
 	void runExclusively(Runnable runnable);
+
+	/**
+	 * Exception thrown when {@link TaskMailbox} is closed.
+	 */
+	class MailboxClosedException extends RejectedExecutionException {
```

Comment:
I'd rather extend `IllegalStateException` and then translate that to a `RejectedExecutionException` in `executor#execute`. It doesn't make sense to throw an `RejectedExecutionException` on `#quiesce` or `#yield`.